pallets/asset-manager/src/lib.rs

```diff
@@ -394,6 +479,99 @@ pub mod pallet {
 		pub fn account_id() -> T::AccountId {
 			T::PalletId::get().into_account()
 		}
+
+		/// Get para id from asset location
+		pub(crate) fn get_para_id_from_multilocation(
+			location: Option<&MultiLocation>,
+		) -> Option<ParaId> {
+			if let Some(MultiLocation { interior, .. }) = location {
+				match interior {
+					// We have some locations like (1, X1(Parachain)).
+					Junctions::X1(Junction::Parachain(para_id)) => Some(*para_id),
+					// We have some locations like (1, X2(Parachain, GeneralKey))
+					// and (1, X2(Parachain, PalletInstance)).
+					Junctions::X2(Junction::Parachain(para_id), ..) => Some(*para_id),
+					// Currently we don't have any Junctions whose length is bigger than 2.
+					_ => None,
+				}
+			} else {
+				None
+			}
+		}
+
+		/// Update the count of associated assets for the para id.
+		pub(crate) fn update_count_of_associated_assets(para_id: ParaId) -> DispatchResult {
```

Comment:
minor: suggested name `increase_count_of_associated_assets`, since this function only increases the count.